core/src/main/java/com/opentable/logging/JsonLogEncoder.java

```diff
@@ -59,31 +58,14 @@ public JsonLogEncoder() {
                 .configure(Feature.AUTO_CLOSE_TARGET, false);
     }
 
-    public void setCustomEventClass(String customEventClass) throws ClassNotFoundException {
```

Comment:
While it's not relevant to this PR the above comment (ObjectMapper) could be resolved by having a non spring singleton with an @Bean bridge. put that in a common dependency and you would solve it?

Reply:
The `JsonLogEncoder` is constructed by Logback entirely outside of the Spring context, so it will never get injected, and must be ready for use before Spring initialization completes.